src/gui/previewselectdialog.cpp

```diff
@@ -93,12 +93,12 @@ PreviewSelectDialog::PreviewSelectDialog(QWidget *parent, const BitTorrent::Torr
     const QVector<qreal> fp = torrent->filesProgress();
     for (int i = 0; i < torrent->filesCount(); ++i)
     {
-        const QString fileName = Utils::Fs::fileName(torrent->filePath(i));
+        const Path fileName = torrent->filePath(i).filename();
         if (Utils::Misc::isPreviewable(fileName))
         {
             int row = m_previewListModel->rowCount();
             m_previewListModel->insertRow(row);
-            m_previewListModel->setData(m_previewListModel->index(row, NAME), fileName);
+            m_previewListModel->setData(m_previewListModel->index(row, NAME), fileName.data());
```

Comment:
> IMO `toString()` is more accurate even if it doesn't contain path separators.

Ok, if you don't care about the excessive search for separators that it will cause.